### PR TITLE
AF-1408: System Properties not taking priority over existing settings for ApplicationPreferencesLoader implementations

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins execute full downstream build

File: uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/config/AppConfigServiceImpl.java [hunk 1]
```diff
@@ -54,28 +60,51 @@ public void configureOnEvent(@Observes ApplicationStarted applicationStartedEven
...
+    }
+
+    private void loadSystemProperties() {
+        for (Map.Entry<String, String> entry : preferences.entrySet()) {
+            final String key = entry.getKey();
+            final String value = entry.getValue();
+            final String newValue = System.getProperty(key, value);
+
+            if (value != null && !value.equals(newValue)) {
+                preferences.put(key, newValue);
```
Comment (jhrcek) [inline uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/config/AppConfigServiceImpl.java]: Hello @Paulovmr,
The code looks and works fine. I only have one small objection about logging consistency. 

The DefaultApplicationPreferencesLoader logs the following on container startup:

```
09:17:33,311 INFO  [org.kie.workbench.common.services.backend.preferences.KieMetaDataPreferencesLoader] (MSC service thread 1-4) Setting preference 'kie_productized' to 'false'.
09:17:33,313 INFO  [org.kie.workbench.common.screens.examples.backend.server.ExamplesPreferencesLoader] (MSC service thread 1-4) Setting preference 'org.kie.demo' to 'true'.
09:17:33,344 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'designer.context' to 'designer'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'designer.profile' to 'jbpm'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.defaultlanguage' to 'en'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'build.enable-incremental' to 'true'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'support.runtime.deploy' to 'true'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.dateformat' to 'dd-MMM-yyyy'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.datetimeformat' to 'dd-MMM-yyyy HH:mm:ss'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'rule-modeller-onlyShowDSLStatements' to 'false'.
09:17:33,345 INFO  [org.guvnor.common.services.backend.config.DefaultApplicationPreferencesLoader] (MSC service thread 1-4) Setting preference 'drools.defaultcountry' to 'US'.
```

Don't you think we should add something similar to line 95 here to make it visible that some of the properties are being overriden?
Something like
"Overriding preference 'X' based on value from system property 'Y'."

Comment (paulovmr) [reply uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/config/AppConfigServiceImpl.java]: @jhrcek I agree, done. Thanks!

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Pull request merged**